jvector-base/src/main/java/io/github/jbellis/jvector/graph/GraphIndexBuilder.java

```diff
@@ -356,8 +356,11 @@ public void cleanup() {
             return;
         }
 
-        // improve connections on everything in L1
-        if (graph.getMaxLevel() > 0) {
+        if (dimension == 2 && graph.getMaxLevel() > 0) {
```

Comment:
Should we consider using KD trees for 2D cases? Just wondering if it's worth making a special case for that.

Reply:
Eventually, we should stop supporting 2d vectors (remove the soecial case) and use a more appropriate data structure (a quadtree, kdtree, etc.). Until we have that, I would keep this.